### ISSUE TITLE
Implement different resolution support

### ISSUE THREAD

**Issue by lookeypl** [mkulagowski/MineZPRft#14]
Right now Game Manager launches the game in default 800x600 window. This definitely needs to be changed.

Support following scenarios:
- [x] Startup set resolution
  - [x] Launching in different resolution (maybe command line arg --resolution/-r followed by two integer values? Lack of the argument would create a window with default 800x600)
  - [ ] Launching in fullscreen (--fullscreen/-f, launches in native )
- [x] Dynamic switching when application is working
  - [x] Switching to fullscreen (Alt+Enter is a default combination, so we should use it)
  - [x] Resizing the window to new resolution by dragging its borders (it works on WindowManager part, but OpenGL does not resize it internally; new resolution values must go to glViewport function and to perspective matrix in shader, which might require minor reorganization in Renderer).

There is no easy way to test it, so we will skip tests for now.

**Comment by mkulagowski** [comment]
[First commit](https://review.gerrithub.io/#/c/258613/) is on. It covers `Dynamic switching when application is working`.
It's only for Windows yet, but when concept is accepted, it will be converted to "both-platforms-commit".

**Comment by mkulagowski** [comment]
Everything apart from fullscreen option is implemented.
Resolution works without '-r' or '--resolution' - just 2 numeric args, if they're incorrect, default res of 800x600 is being used.

I guess that the fullscreen option should work with and/or without resolution option.

Maybe we should introduce some args parser, that would make adding argument option easy?
I'm talking about being able to add short and/or long version, verification function and an argument callback.
Here's example of my vision:

```
short: -r
long: --resolution
arg number: 2
verification func: [](auto w, auto h)->bool{ return IsNumeric(w) && IsNumeric(h);};
argument callback: GameManager::GetInstance().GetWindow()->SetSize();
```

all of this should somehow be added - either dynamically in the main.cpp before some ArgumentParser::Parse() method or statically in ArgumentParser class header.

I guess more would become known, if I were to actually start implementing this.
Whaddya think @LKostyra ?

**Comment by lookeypl** [comment]
Indeed, this might be a good idea. It will make adding new arguments easier.

Since you somehow must add the list of arguments to the Parser, I suggest implementing an AddArgument() method, which will return reference to this. In this way we will be able to easily chain argument addition code into one call:

```
Parser::AddArgument(...)
       .AddArgument(...)
       .AddArgument(...)
       //...
```

What do you think?

**Comment by mkulagowski** [comment]
I think this: #18.
